scalding-parquet-scrooge/src/main/scala/com/twitter/scalding/parquet/scrooge/PartitionedParquetScroogeSource.scala

```diff
@@ -0,0 +1,48 @@
+package com.twitter.scalding.parquet.scrooge
+
+import _root_.cascading.scheme.Scheme
+import _root_.cascading.tuple.Fields
+import com.twitter.scalding._
+import com.twitter.scalding.parquet.thrift.ParquetThriftBase
+import com.twitter.scalding.typed.{ PartitionSchemed, PartitionUtil }
+import com.twitter.scrooge.ThriftStruct
+
+/**
+ * Scalding source to read or write partitioned Parquet scrooge data.
+ *
+ * For writing it expects a pair of `(P, T)`, where `P` is the data used for partitioning and
+ * `T` is the scrooge object. `P` must be either a String or a tuple of Strings.
+ * Below is an example.
+ * {{{
+ * val data: TypedPipe[MyScroogeObject] = ???
+ * data.map { obj =>
+ *   ( (obj.country, obj.city), obj)
+ * }.write(PartitionedParquetScroogeSource[(String, String), MyScroogeObject](path, "%s/%s"))
+ * }}}
+ *
+ * For reading it produces a pair `(P, T)` where `P` is the partition data, `T` is the corresponding
+ * scrooge object. Below is an example.
+ * {{{
+ * val in: TypedPipe[(String, String), MyScroogeObject] =
+ * TypedPipe.from( PartitionedParquetScroogeSource[(String, String), MyScroogeObject](path, "%s/%s") )
+ * }}}
+ *
+ */
+case class PartitionedParquetScroogeSource[P, T <: ThriftStruct](
+  path: String, template: String, fields: Fields = PartitionUtil.toFields(0, implicitly[TupleSetter[T]].arity))(implicit val mf: Manifest[T],
```

Comment:
Yeah, any new code should certainly not expose it. How much to remove/lock
down of old code is a question.
On Fri, Sep 9, 2016 at 14:55 Piyush Narang <notifications@github.com> wrote:

> In
> scalding-parquet-scrooge/src/main/scala/com/twitter/scalding/parquet/scrooge/PartitionedParquetScroogeSource.scala
> <https://github.com/twitter/scalding/pull/1590#discussion_r78267367>:
>
> > + * val data: TypedPipe[MyScroogeObject] = ???
> > + * data.map { obj =>
> > + *   ( (obj.country, obj.city), obj)
> > + * }.write(PartitionedParquetScroogeSource[(String, String), MyScroogeObject](path, "%s/%s"))
> > + * }}}
> > + *
> > + * For reading it produces a pair `(P, T)` where `P` is the partition data, `T` is the corresponding
> > + * scrooge object. Below is an example.
> > + * {{{
> > + * val in: TypedPipe[(String, String), MyScroogeObject] =
> > + * TypedPipe.from( PartitionedParquetScroogeSource[(String, String), MyScroogeObject](path, "%s/%s") )
> > + * }}}
> > + *
> > + */
> > +case class PartitionedParquetScroogeSource[P, T <: ThriftStruct](
> > +  path: String, template: String, fields: Fields = PartitionUtil.toFields(0, implicitly[TupleSetter[T]].arity))(implicit val mf: Manifest[T],
>
> Yeah can set Fields to that. Can override it in
> PartitionedParquetScroogeSource and PartitionedParquetThriftSource unless
> you'd rather have it set in PartitionSchemed and not exposed to
> PartitionedParquet{Scrooge|Thrift}Source and PartitionedDelimitedSource?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/twitter/scalding/pull/1590/files/363fed0f3820cb03a8faf8cf54a127b1f1f399f8#r78267367>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEJdul73q_uAfVf2X0ZD2EJMHEO82N3ks5qof_vgaJpZM4Jw9Ah>
> .
>